hspec-core/src/Test/Hspec/Core/Spec.hs

```diff
@@ -44,8 +51,23 @@ describe label spec = runIO (runSpecM spec) >>= fromSpecList . return . specGrou
 -- > describe "absolute" $ do
 -- >   it "returns a positive number when given a negative number" $
 -- >     absolute (-1) == 1
-it :: Example a => String -> a -> SpecWith (Arg a)
-it label action = fromSpecList [specItem label action]
+it :: (?loc :: CallStack, Example a) => String -> a -> SpecWith (Arg a)
+it label action = do
+  cwd <- runIO getCurrentDirectory
+  addLocation cwd $ fromSpecList [specItem label action]
+  where
+    addLocation cwd = mapSpecItem_ (\item -> item { itemLocation = toLocation cwd <$> srcLoc })
+
+    -- The IP type signatures on ?loc here and on ‘it’ itself as well
+    -- as the actual IP names are *essential*: if the names differ or
+    -- one of the signatures is missing, IP scoping rules or CallStack
+    -- rules will mess the stack up and we won't get what we wanted.
+    srcLoc = case reverse (getCallStack (?loc :: (?loc :: CallStack) => CallStack)) of
+      ("it", loc):_ -> Just loc
+      _ -> Nothing
+
+    toLocation cwd s =
+      Location (makeRelative cwd $ srcLocFile s) (srcLocStartLine s) (srcLocStartCol s) ExactLocation
```

Comment:
Using the binary from @nh2 (https://github.com/nh2/ghc/releases/tag/ghc-7.10.1-release-srcloc-ip), I get relative paths.  From my perspective it's desirable for paths in source to follow the exact same semantics as the ones in regular compile error messages.

@Fuuzetsu can you try to find out whether the code changed in the meanwhile, or is it due to your build environment?  E.g. can you compare the format of compile errors and src locations in your environment?